src/java/org/apache/cassandra/stress/settings/AuthProvider.java

```diff
@@ -0,0 +1,59 @@
+package org.apache.cassandra.stress.settings;
+
+import org.apache.cassandra.stress.util.JavaDriverV4Config;
+import org.apache.cassandra.stress.util.JavaDriverV4Session;
+import shaded.com.datastax.oss.driver.api.core.CqlSessionBuilder;
+import shaded.com.datastax.oss.driver.api.core.config.ProgrammaticDriverConfigLoaderBuilder;
+
+public class AuthProvider {
+  String authClassName;
+  String username;
+  String password;
+
+  public AuthProvider(String authClassName, String username, String password) {
+    this.authClassName = authClassName;
+    this.username = username;
+    this.password = password;
+  }
+
+  public String ClassName() {
+    return authClassName;
+  }
+
+  public com.datastax.driver.core.AuthProvider ToV3() {
+    if (authClassName == null || authClassName.isEmpty()) {
+      return null;
+    }
+
+    try {
+      if (authClassName.equals("com.datastax.driver.core.PlainTextAuthProvider") || authClassName.equals("PlainTextAuthProvider")) {
+        return new com.datastax.driver.core.PlainTextAuthProvider(username, password);
+      }
+      throw new IllegalArgumentException("Unknown auth provider class: " + authClassName);
+    } catch (IllegalArgumentException e) {
+      throw e;
+    } catch (Exception e) {
+      throw new IllegalArgumentException("Failed to initialize authentication class: " + authClassName, e);
+    }
+  }
+
+  public JavaDriverV4Session ToV4() {
+    if (authClassName == null || authClassName.isEmpty()) {
+      return null;
+    }
+    if (authClassName.equals("com.datastax.oss.driver.internal.core.auth.PlainTextAuthProvider") ||
+        authClassName.equals("PlainTextAuthProvider") ||
+        authClassName.equals("com.datastax.oss.driver.api.core.auth.ProgrammaticPlainTextAuthProvider") ||
+        authClassName.equals("ProgrammaticPlainTextAuthProvider")
+    ) {
+      return new JavaDriverV4Session() {
```

Comment:
nit: not really a driver session, but something that customizes CqlSessionBuilder.